api/openapi-spec/swagger.json

```diff
@@ -23561,7 +23561,7 @@
         ],
         "x-kubernetes-action": "connect",
         "x-kubernetes-group-version-kind": {
-          "group": "",
+          "group": "Pod",
```

Comment:
This file is generated. To make the tests pass, you need to change the generation steps. I assume that you want to customize the `x-kubernetes-group-version-kind` when the action is `CONNECT`. You probably want to change the code [here](https://github.com/kubernetes/kubernetes/blob/deebe6619a1b4955b0cd109a8f05b1298a260310/staging/src/k8s.io/apiserver/pkg/endpoints/installer.go#L960-L965).

That being said, I haven't understood what the fix does yet. Could you update the PR description to include: which exact issue (among the issues in https://github.com/kubernetes/kubernetes/issues/95052) this PR fixes, and how this change fixes the issue?

Reply:
Thanks @roycaihw for the tip. I was trying to add the group info for missed GVKs as per https://github.com/Arnavion/k8s-openapi/blob/master/k8s-openapi-codegen/src/fixups/upstream_bugs.rs

And @dims seems i have to change the code as indicated by @roycaihw [here](https://github.com/kubernetes/kubernetes/blob/deebe6619a1b4955b0cd109a8f05b1298a260310/staging/src/k8s.io/apiserver/pkg/endpoints/installer.go#L960-L965). Because running hack/update-openapi-spec.sh is creating the same old swagger spec with missing group details.

Reply:
@gkarthiks I didn't find a missing-group issue. Are you trying to fix [connect_options_gvk](https://github.com/Arnavion/k8s-openapi/blob/0b986e156511f5d4dcf7577cf3905aaa8641cf0c/k8s-openapi-codegen/src/fixups/upstream_bugs.rs#L5-L9)? If so, the issue is that the "kind" is wrong.

Reply:
can you provide any guidance on how to fix this? Perhaps I misunderstood @dims issue https://github.com/kubernetes/kubernetes/issues/95052. Any help on this from @dims or @roycaihw is highly appreciated.

Reply:
@nikhita perhaps?

Reply:
@gkarthiks Regarding `connect_options_gvk`, you can see my comments in https://github.com/kubernetes/kubernetes/issues/95052#issuecomment-702899811 and https://github.com/kubernetes/kubernetes/issues/95052#issuecomment-702988635. The parent kinds should be used when generating the spec.